File: DIferencias_finitas-1D/Función.jl
function analitica(x)
    cos(x)*(1-x/2)-sin(x)
end


function coef1()
    1.
end

function coef2(x)
    sin(x)
end

function thomas(A,B,u)
    n=length(A[:,1])
    for i in Iterators.countfrom(1,1)
        i>n-1 && break
        piv=-A[i+1,1]/A[i,2]
        A[i+1,2]=piv*A[i,3]+A[i+1,2]
        B[i+1]=piv*B[i]+B[i+1]
    end
    u[n]=B[n]/A[n,2]
    for i in Iterators.countfrom(n-1,-1)
        i<1 && break
        u[i]=(B[i]-A[i,3]*u[i+1])/A[i,2]
    end
end

File: DIferencias_finitas-1D/Python/Función.py
import numpy as np 

def coef1():
    c=1
    return c

def coef2(x):
    f=np.sin(x)
    return f

def analitica(x):
    U=np.cos(x)*(1-x/2)-np.sin(x)
    return U

def mat_trid(matriz,i,j,v):
    d=j-i
    j=d+1
    matriz[i,j]=v

def thomas(a,b):
    n=a.shape[0]
    for i in range(n-1):
        piv=-a[i+1,0]/a[i,1]
        a[i+1,1]=piv*a[i,2]+a[i+1,1]
        b[i+1]=piv*b[i]+b[i+1]
    u=np.zeros((n),dtype=np.float64)
    u[n-1]=b[n-1]/a[n-1,1]
    i=n-2
    while i>=0:
        u[i]=(b[i]-a[i,2]*u[i+1])/a[i,1]
        i=i-1
    return u
